### ISSUE TITLE
stateful switch is not switched on, only switched off

### ISSUE THREAD

**Issue by magune** [hugobloem/stateful_scenes#106]
I love the concept of this integration, but i can't make it work.

I have 3 buttons to activate 3 scenes. They all activate the stateful switch, rather than activating the scene itself. clicking any of the buttons do turn on/off the lights correctly. But clicking a button to activate the scene never turns the state to ON. But clicking any of the other buttons do turn it OFF.

I have Philips Hue and one Ikea light.

It may be due to the default tolerance setting set to default = 1. I feel quite stupid at this point, but I find no place to change the tolerance. That settings just isnt visible anywhere I can reasonably look for it:
<img width="351" alt="image" src="https://github.com/user-attachments/assets/ad6bb236-c938-450a-8659-ee3884f5dc49">

**Comment by pgale** [comment]
I'm super keen to have this working but the Hue functions just aren't working. Unfortunately the developer hasn't been on here for months so I've given up on it for now. Shame as it would be amazing if it worked.

**Comment by magune** [comment]
#pgale, agreed. Unfortunately I am not a developer so I can't fix code myself, also I assume the maintainer must accept code contributions anyway.

**Comment by magune** [comment]
@pgale , do you know where to change the tolerance setting? i cant find it anywhere.

**Comment by pgale** [comment]
Yeah me too. Could be forked though?

**Comment by pgale** [comment]
Not off the top of my head and I've uninstalled it. Isn't it a setting in the integration's UI?

**Comment by magune** [comment]
I saw the setting at installation time but not after

**Comment by magune** [comment]
i have now managed to find the tolerance setting. if i try to add the integration again, i get the config dialog and it says it is saving the increased tolerance. Increasing to 2-3 made my 3 buttons with 3 scenes (off, cosy, normal) for room 1 to work. but for room 2, 3 similar buttons for 3 similar scenes still do not work, e.g. other scenes turn off correct when another scene is triggered, but the stateful scene do not turn on correctly when activated. but in all cases, the scenes are activated correctly.

**Comment by hugobloem** [comment]
Hi, thank you for your interest in this integration and for taking the time to open this issue. This integration is something I try to work on in my spare time and this has been limited lately. 

I am trying to understand your issue, but am I correct in thinking that most issues seem resolved when increasing the tolerance to ~3 rather than the default 1?

**Comment by SkywiperSolutions** [comment]
Same issue here:

Scenes activate and turns off, but entity state of the Stateful Scene entity never changes to on - only off.
I tried different configuration options with up to 10 sec debound, tolerance 5 etc. No difference.

**Comment by magune** [comment]
> Hi, thank you for your interest in this integration and for taking the time to open this issue. This integration is something I try to work on in my spare time and this has been limited lately.
> 
> I am trying to understand your issue, but am I correct in thinking that most issues seem resolved when increasing the tolerance to ~3 rather than the default 1?

i have tried 2.3 and also the maximum tolerance (I think it was 9 or something like that), it didnt help. but I don't know if the changed tolerance settings actually are saved and used, or if the gui accepts the change and it doesnt affect the logic thereafter.

**Comment by cayossarian** [comment]
I had an issue where the stateful switch scene would turn on momentarily and then off.  When I looked at the yaml for the scene and what the scene was actually producing when turned on in the light they were off by an amount which was resolved by deleting the configuration for the integration as a whole and then adding the entry again, but changing the rounding to 3 in the main configuration.  Also had to have a transition on that dimmer of 1 or 2.    

The rounding might be better done at the device level just like transition and debounce.   If the repository owner agrees I'd be willing to make a PR.
